garage/tf/models/gaussian_mlp_model.py

```diff
@@ -0,0 +1,220 @@
+"""Gaussian MLP Model."""
+import numpy as np
+import tensorflow as tf
+
+from garage.core import Serializable
+from garage.misc import ext
+from garage.misc.overrides import overrides
+from garage.tf.core.networks import mlp, parameter
+from garage.tf.models import Model
+
+
+class GaussianMLPModel(Model, Serializable):
+    """Gaussian MLP Model."""
+
+    def __init__(
+            self,
+            input_dim,
+            output_dim,
+            name="GaussianMLPModel",
+            hidden_sizes=(32, 32),
+            learn_std=True,
+            init_std=1.0,
+            adaptive_std=False,
+            std_share_network=False,
+            std_hidden_sizes=(32, 32),
+            min_std=1e-6,
+            max_std=None,
+            std_hidden_nonlinearity=tf.nn.tanh,
+            hidden_nonlinearity=tf.nn.tanh,
+            output_nonlinearity=None,
+            std_parameterization='exp',
+    ):
+        """
+        Initialize a gaussian mlp model and build the graph.
+
+        :param input_dim: input dimension
+        :param output_dim: output dimension
+        :param name: name of the model
+        :param hidden_sizes: list of sizes for the fully-connected hidden
+          layers
+        :param learn_std: Is std trainable
+        :param init_std: Initial std
+        :param adaptive_std:
+        :param std_share_network:
+        :param std_hidden_sizes: list of sizes for the fully-connected layers
+          for std
+        :param min_std: whether to make sure that the std is at least some
+          threshold value, to avoid numerical issues
+        :param std_hidden_nonlinearity:
+        :param hidden_nonlinearity: nonlinearity used for each hidden layer
+        :param output_nonlinearity: nonlinearity for the output layer
+        :param mean_network: custom network for the output mean
+        :param std_network: custom network for the output log std
+        :param std_parametrization: how the std should be parametrized. There
+          are a few options:
+            - exp: the logarithm of the std will be stored, and applied a
+                exponential transformation
+            - softplus: the std will be computed as log(1+exp(x))
+        :return:
+        """
+        Serializable.quick_init(self, locals())
+        super(GaussianMLPModel, self).__init__()
+
+        self.name = name
+        self._variable_scope = tf.variable_scope(
+            self.name, reuse=tf.AUTO_REUSE)
+        self._name_scope = tf.name_scope(self.name)
+
+        # Network parameters
+        self._input_dim = input_dim
+        self._output_dim = output_dim
+        self._hidden_sizes = hidden_sizes
+        self._learn_std = learn_std
+        self._init_std = init_std
+        self._adaptive_std = adaptive_std
```

Comment:
adaptive_std and std_share_network are different architectures so IMO they should be different models